milatools/utils.py

```diff
@@ -1,154 +1,4 @@
-        else:
-            print("# OK")
-            return True
-
-
-class SSHConnection:
-    def __init__(self, host):
-        self.here = Local()
-        os.makedirs(sockdir, mode=0o700, exist_ok=True)
-        self.host = host
-        self.sock = os.path.join(sockdir, f"milatools.{host}")
-        self.master = self.here.popen(
-            "ssh",
-            host,
-            "-fNMS",
-            self.sock,
-            stdout=subprocess.PIPE,
-            stderr=subprocess.STDOUT,
-        )
-
-    def cmd(self, *args, bash=False):
-        if bash:
-            args = [shlex.join(["bash", "-c", *args])]
-        return ["ssh", self.host, "-S", self.sock, *args]
-
-    def display(self, args):
-        print(T.bold_cyan(f"({self.host}) $ ", *args))
-
-    def get(self, *args, bash=False):
-        self.display(args)
-        cmd = self.cmd(*args, bash=bash)
-        return subprocess.check_output(
-            cmd,
-            universal_newlines=True,
-        )
-
-    def popen(self, *args, bash=False):
-        self.display(args)
-        cmd = self.cmd(*args, bash=bash)
-        return subprocess.Popen(
-            cmd,
-            stdout=subprocess.PIPE,
-            stderr=subprocess.STDOUT,
-            universal_newlines=True,
-        )
-
-    def extract(self, *args, pattern, wait=False, bash=False):
-        proc = self.popen(*args, bash=bash)
-        result = None
-        try:
-            while line := proc.stdout.readline():
-                print("#", line.rstrip())
-                if m := re.match(pattern, line):
-                    result = m.groups()[0]
-                    if not wait:
-                        return proc, result
-        except KeyboardInterrupt:
-            proc.terminate()
-            exit("Canceled")
-        proc.wait()
-        return None, result
-
-    def wait(self):
-        self.master.wait()
-
-    def cleanup(self):
-        pass
-
-
-def yn(question, default="y"):
-    """Ask a yes/no question."""
-    options = "[y/n]".replace(default, default.upper())
-    answer = input(T.bold(f"{question} {options} ")).strip()
-    return (answer or default) in "yY"
-
-
-class SSHConfig:
-    """Wrapper around sshconf with some extra niceties."""
-
-    def __init__(self, path):
-        self.cfg = read_ssh_config(path)
-        self.add = self.cfg.add
-        self.save = self.cfg.save
-        self.host = self.cfg.host
-        self.hosts = self.cfg.hosts
-
-    def hoststring(self, host):
-        lines = []
-        for filename, cfg in self.cfg.configs_:
-            lines += [line.line for line in cfg.lines_ if line.host == host]
-        return "\n".join(lines)
-
-    def confirm(self, host):
-        print(T.bold("The following code will be appended to your ~/.ssh/config:\n"))
-        print(self.hoststring(host))
-        return yn("\nIs this OK?")
+running_on_mila_cluster = socket.getfqdn().endswith(".server.mila.quebec") and "SLURM_TMPDIR" in os.environ.keys()
```

Comment:
I think it would be useful to be able to optimize dataset loading on other clusters/environments than Mila's, so I think it would be better to have a `ComputeLocation` or `ComputeEnvironment` class, with subclasses like `MilaEnvironment`, `LocalEnvironment`, `BelugaEnvironment`, etc. We can have a `get_compute_environment()` function to get the proper environment.